pkgs/development/python-modules/frida-python/default.nix

```
@@ -0,0 +1,39 @@
+{ lib, stdenv, fetchurl, fetchPypi, buildPythonPackage, typing-extensions }:
+let
+  version = "16.0.19";
+  devkit = fetchurl {
+    url = "https://github.com/frida/frida/releases/download/${version}/frida-core-devkit-${version}-linux-x86_64.tar.xz";
+    hash = "sha256-yNXNqv8eCbpdQKFShpAh6rUCEuItrOSNNLOjESimPdk=";
+  };
+in buildPythonPackage rec {
+  pname = "frida-python";
+  inherit version;
+
+  src = fetchPypi {
+    pname = "frida";
+    inherit version;
+    hash = "sha256-rikIjjn9wA8VL/St/2JJTcueimn+q/URbt9lw/+nalY=";
+  };
+
+  postPatch = ''
+    mkdir assets
+    pushd assets
+    tar xvf ${devkit}
+    export FRIDA_CORE_DEVKIT=$PWD
+    popd
+  '';
+
+  propagatedBuildInputs = [ typing-extensions ];
+
+  pythonImportsCheck = [ "frida" ];
+
+  passthru = { inherit devkit; };
+
+  meta = with lib; {
+    description = "Dynamic instrumentation toolkit for developers, reverse-engineers, and security researchers (Python bindings)";
+    homepage = "https://www.frida.re";
+    license = licenses.wxWindows;
+    maintainers = with maintainers; [ s1341 ];
+    platforms = [ "x86_64-linux" ];
+  };
+}
```

```suggestion
  meta = {
    description = "Dynamic instrumentation toolkit for developers, reverse-engineers, and security researchers (Python bindings)";
    homepage = "https://www.frida.re";
    license = lib.licenses.wxWindows;
    maintainers = with lib.maintainers; [ s1341 ];
    platforms = [ "x86_64-linux" ];
  };
}
```

For some background see:
- https://nix.dev/recipes/best-practices#with-scopes
- https://github.com/NixOS/nixpkgs/pull/229030

---

We're far from consensus on this.  I don't think it's fair to force contributors to adopt one style or the other when both are still commonly used and advocated for.

---

While we don't have consensus on `with` I think getting less nested `with`'s is a good thing that should be fairly uncontroversial.
You'll note that `with` is still used in the suggested code, just localised.

What we have now is a situation where everyone just blindly copy & paste expressions assuming that whatever else they can find in nixpkgs follows good practice.

I don't think we need consensus to suggest not adding more code smells.

---

I fixed this.